core/src/main/scala/io/finch/request/package.scala

```diff
@@ -162,6 +180,8 @@ package object request {
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) =
         try number.toFuture
         catch { case _: IllegalArgumentException => ValidationFailed(param, rule).toFutureException }
+
+      override def toString: String = s"param: $param"
```

Comment:
Yes, I could take over, I wanted to do some work over the weekend anyway (two more proposals regarding RequestReader, one of them potentially controversial), so I could add validation to that. But I feel we almost could have merged this PR first (I think I could build the reusable rules on top).

Did we settle with renaming `should` to `which`? I think I don't have a strong opinion on the name.